homeassistant/components/roborock/device.py

```diff
@@ -21,27 +22,24 @@ class RoborockCoordinatedEntity(CoordinatorEntity[RoborockDataUpdateCoordinator]
     def __init__(
         self,
         unique_id: str,
-        device_info: RoborockHassDeviceInfo,
         coordinator: RoborockDataUpdateCoordinator,
     ) -> None:
         """Initialize the coordinated Roborock Device."""
         super().__init__(coordinator)
         self._attr_unique_id = unique_id
-        self._device_name = device_info.device.name
-        self._device_id = device_info.device.duid
-        self._device_model = device_info.product.model
-        self._fw_version = device_info.device.fv
+        self._device_name = coordinator.device_info.device.name
+        self._device_id = coordinator.device_info.device.duid
+        self._device_model = coordinator.device_info.product.model
+        self._fw_version = coordinator.device_info.device.fv
```

Comment:
Optional style suggestion: I might also suggest to just inline these directly in `device_info` below, getting the fields right from the coordinator just to save some vertical space since we're holding on to the coordinator already

Reply:
I agree. Made that change. One point is that I created a property to more easily access device_id from within DeviceInfo